pkg/dhcp6client/client.go

```diff
@@ -313,15 +319,18 @@ func (c *Client) newClientErr(err error) *ClientError {
 // TODO(hugelgupf): since the client only has one connection, maybe it should
 // just have one dedicated goroutine for reading from the UDP socket, and use a
 // request and response queue.
-func (c *Client) SimpleSendAndRead(ctx context.Context, dest *net.UDPAddr, p *dhcp6.Packet) (<-chan *ClientPacket, <-chan *ClientError) {
+func (c *Client) SimpleSendAndRead(ctx context.Context, dest *net.UDPAddr, p *dhcp6.Packet) (*sync.WaitGroup, <-chan *ClientPacket, <-chan *ClientError) {
```

Comment:
I would have rather taken a waitgroup as argument, to be able to pool it between multiple independent clients (eg v4/v6, multiple interfaces)

Reply:
To pool multiple clients, one should use SendAndRead instead of the simple version anyway.